android/app/src/main/java/network/perun/app/MainActivity.java

```diff
@@ -198,7 +198,25 @@ public void handleUpdate(ChannelUpdate update, UpdateResponder responder) {
             Log.i("channel", String.format("Update (version=%d, isFinal=%b)", state.getVersion(), state.isFinal()));
             responder.accept(ctx);
             BigInts bals = update.getState().getBalances();
-            Log.d("channel", String.format("Acceped update (version=%d, bals=[%s, %s])", state.getVersion(), bals.get(update.getActorIdx()).toString(), bals.get(1-update.getActorIdx()).toString()));
+            Log.d("channel", String.format("Accepted update (version=%d, bals=[%s, %s])", state.getVersion(), bals.get(update.getActorIdx()).toString(), bals.get(1-update.getActorIdx()).toString()));
+        } catch (Exception e) {
+            Log.e("channel", e.toString());
+        } finally {
+            ctx.cancel();
+        }
+    }
+
+    @Override
+    public void handleConcluded(byte[] id) {
+        Log.i("channel", "Received concluded event for channel " + new BigInteger(1, id).toString(16));
+        Context ctx = Prnm.contextWithTimeout(30);
+        try {
+            PaymentChannel ch = chs.get(ByteBuffer.wrap(id));
+            if (ch == null) {
+                throw new Exception(String.format("handleConcluded: channel not found id=%s", new BigInteger(1, id).toString(16)));
+            }
+            ch.settle(ctx, true);
```

Comment:
In this case both parties would react to the `concludedEvent` by calling settle. Since this is example code i think it is ok, but maybe add a comment.

Reply:
You're right, I'll change it so it returns in this case, similarly to the demo.